lib/mongo/retryable.rb

```diff
@@ -46,18 +46,19 @@ module Retryable
     # @return [ Result ] The result of the operation.
     #
     # @since 2.1.0
-    def read_with_retry(attempt = 0, &block)
+    def read_with_retry(attempt = 0, max_read_retries = nil, &block)
+      max_retries = max_read_retries || cluster.max_read_retries
       begin
         block.call
       rescue Error::SocketError, Error::SocketTimeoutError
         retry_operation(&block)
       rescue Error::OperationFailure => e
         if cluster.sharded? && e.retryable?
```

Comment:
`OperationFailure` here would raise a `NoMethodError` on including classes that don't have a reference to `Cluster`. I think to make this module completely reusable we should remove all references to `cluster` or make another more generic retry module for the monitor and connection.